detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/BracesOnWhenStatements.kt

```diff
@@ -0,0 +1,254 @@
+
+    @Configuration("multi-line braces policy")
+    private val multiLine: BracePolicy by config("consistent") { BracePolicy.getValue(it) }
+
+    override fun visitWhenExpression(expression: KtWhenExpression) {
+        super.visitWhenExpression(expression)
+
+        val branches: List<KtExpression> = walk(expression)
+        validate(branches, policy(expression))
+    }
+
+    private fun walk(expression: KtWhenExpression): List<KtExpression> {
+        return expression.entries.mapNotNull { it.expression }
+    }
+
+    private fun validate(list: List<KtExpression>, policy: BracePolicy) {
+        val violators = when (policy) {
+            BracePolicy.Always -> {
+                list.filter { !hasBraces(it) }
+            }
+
+            BracePolicy.Necessary -> {
+                list.filter { !isMultiStatement(it) && hasBraces(it) }
+            }
+
+            BracePolicy.Never -> {
+                list.filter { hasBraces(it) }
+            }
+
+            BracePolicy.Consistent -> {
+                val braces = list.count { hasBraces(it) }
+                val noBraces = list.count { !hasBraces(it) }
+                if (braces != 0 && noBraces != 0) {
+                    list.take(1)
+                } else {
+                    emptyList()
+                }
+            }
+        }
+        violators.forEach { report(it, policy) }
+    }
+
+    private fun hasBraces(expression: KtExpression): Boolean =
+        expression is KtBlockExpression
+
+    private fun KtExpression.parentCandidate(): PsiElement? =
+        this.parent.parent
+
+    private fun isMultiStatement(expression: KtExpression): Boolean =
+        expression is KtBlockExpression && expression.statements.size > 1
+
+    private fun policy(expression: KtWhenExpression): BracePolicy {
+        val multiLineCandidate = expression.entries.firstOrNull { entry ->
+            entry.text.substringAfter("->").contains('\n') ||
+                entry.expression?.let { isMultiStatement(it) } ?: false
+        }
+        return if (multiLineCandidate != null) multiLine else singleLine
+    }
+
+    private fun report(violator: KtExpression, policy: BracePolicy) {
+        val parent = violator.parentCandidate() as KtWhenExpression
+        val entries = parent.entries.mapNotNull { it.expression }
+        val reported = when {
+            violator in entries && policy == BracePolicy.Consistent -> parent.whenKeyword
+            violator in entries -> violator
+            else -> error("Violating element (${violator.text}) is not part of this 'when' (${parent.text})")
+        }
+        val message = when (policy) {
+            BracePolicy.Always -> "Missing braces on this branch, add them."
+            BracePolicy.Consistent -> "Inconsistent braces, make sure all branches either have or don't have braces."
+            BracePolicy.Necessary -> "Extra braces exist on this branch, remove them (ignore multi-statement)."
```

Comment:
```suggestion
            BracePolicy.Necessary -> "Extra braces exist on this branch, remove them."
```
you can remove this, multi-statement for necessary shouldn't be flagged, right? That's the reason why `necessary` exists.

Reply:
right